datastream-server-restli/src/main/java/com/linkedin/datastream/server/dms/DatastreamResources.java

```diff
@@ -686,6 +686,37 @@ public Datastream get(String name) {
     return stream;
   }
 
+  /**
+   * Finds the task's assignment node which hosts the given task 
+   * @param pathKeys Datastream resource key
+   * @param task Datastream task name
+   */
+  @Action(name = "getTaskAssignment")
+  public String getTaskAssignment(@PathKeysParam PathKeys pathKeys, @ActionParam("task") String task) {
+    // Get datastream
+    String datastreamName = pathKeys.getAsString(KEY_NAME);
+
+    Datastream stream;
+    String assignedInstance = null;
+    try {
+      LOG.info("Get assigned instance for task: {}", task);
+      _dynamicMetricsManager.createOrUpdateMeter(CLASS_NAME, GET_CALL, 1);
+      stream = _store.getDatastream(datastreamName);
```

Comment:
what if the stream does not exists or is in paused state?

Reply:
In paused state, it will return "PAUSED_INSTANCE", and yea I'll put a null check.

Reply:
updated, it throws DatastreamException which then further throws RestLiServiceException from catch block.